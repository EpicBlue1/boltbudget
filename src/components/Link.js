import {useState} from 'react';

export default function Link({page, children}) {

    const [status, setStatus] = useState("Normal");

    const onMouseEnter = () => {
        setStatus("hovered")
    };

    const onMouseLeave = () => {
        setStatus("normal")
    };

    return (
        <a 
        href={page || '#'}
        style={{color: status == "normal" ? "blue" : "green"}}
        onMouseEnter = {onMouseEnter}
        onMouseLeave = {onMouseLeave}

        >
            {children}
        </a>
    )
}