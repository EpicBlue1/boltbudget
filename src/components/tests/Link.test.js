import renderer from 'react-test-renderer';
import Link from '../Link';


//test to see if it rendered correctly
describe('Testing link component', () => {

    test('should', () => {

            //create render component (as you would in the return of a component)
    const component = renderer.create(
        <Link Page="https://facebook.com/">Facebook</Link>
    );

    //render as json to investigate
    let tree = component.toJSON();

    //Test if it rendered correctly
    expect(tree).toMatchSnapshot();

    //2. simulate interaction with component

    //trigger the leave of component hover
    renderer.act(() => {
        tree.props.onMouseEnter();
    })
        
     //re-render tree
     tree = component.toJSON();
     expect(tree).toMatchSnapshot();

    })


})