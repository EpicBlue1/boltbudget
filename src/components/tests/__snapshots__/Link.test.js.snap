// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<a
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "color": "green",
    }
  }
>
  Facebook
</a>
`;

exports[` 2`] = `
<a
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "color": "green",
    }
  }
>
  Facebook
</a>
`;

exports[`Testing link component should 1`] = `
<a
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "color": "green",
    }
  }
>
  Facebook
</a>
`;

exports[`Testing link component should 2`] = `
<a
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "color": "green",
    }
  }
>
  Facebook
</a>
`;
